diff --git a/index.js b/index.js
--- a/index.js
+++ b/index.js
@@ -1,6 +1,8 @@
 const { platform, arch } = process;
 let nativeBinding = null;
 try {
-  if (platform === 'win32') { nativeBinding = require('./triviumdb.win32-x64-msvc.node'); }
+  if (platform === 'win32' && arch === 'arm64') { nativeBinding = require('./triviumdb.win32-arm64-msvc.node'); }
+  else if (platform === 'win32' && arch === 'x64') { nativeBinding = require('./triviumdb.win32-x64-msvc.node'); }
+  else if (platform === 'win32') { throw new Error(`Unsupported TriviumDB Windows architecture: ${arch}`); }
   else if (platform === 'darwin' && arch === 'arm64') { nativeBinding = require('./triviumdb.darwin-arm64.node'); }
   else if (platform === 'darwin') { nativeBinding = require('./triviumdb.darwin-x64.node'); }
